### PR TITLE
KNE-27: Migrate testing framework to jasmine

### PR DESCRIPTION
**Pull request by rossnelson**
- Install jasmine
- Setup a basic testing senario for the master engine process
- Implement some of the previous tests

Well, looks like KNE-29 (Refactor existing tests) is included in this PR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,17 +4,17 @@
   "description": "The Knetik Node Engine, a modular engine written in Javascript.",
   "main": "app.js",
   "scripts": {
-    "vet": "jshint app.js submodules test",
-    "test-init": "NODE_ENV=development node test/test-init/app.js",
-    "test-mocha": "NODE_ENV=testing mocha",
-    "test": "npm run vet; if [ $? -ne 0 ]; then exit 1; fi; npm run test-mocha; if [ $? -ne 0 ]; then exit 1; fi;"
+    "vet": "jshint app.js submodules spec",
+    "unit-test": "NODE_ENV=testing CWD=\"$(pwd)\" jasmine",
+    "test": "npm run vet; if [ $? -ne 0 ]; then exit 1; fi; yarn unit-test; if [ $? -ne 0 ]; then exit 1; fi;"
```
Comment (jgpacheco) [inline package.json]: Don't you like gulp for things like this?

Comment (rossnelson) [inline package.json]: no

File: package.json [hunk 2]
```diff
@@ -27,7 +27,6 @@
   },
   "devDependencies": {
     "jshint": "^2.9.1",
-    "mocha": "^2.4.5",
     "should": "^8.2.2"
```
Comment (jgpacheco) [inline package.json]: `should` is no longer needed if we are using `jasmine`.

File: spec/engine.spec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+global.KNE_BASE_PATH = process.env.CWD + '/spec/support/application';
+
+const engine = require('../app');
+const cluster = require('cluster');
+
+describe('Engine Master', () => {
```
Comment (jgpacheco) [inline spec/engine.spec.js]: Great! I will update [PR#5](https://github.com/knetikmedia/kne/pull/5) with unit tests after approval.

File: spec/support/application/configs/config.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+module.exports = {
```
Comment (rossnelson) [inline spec/support/application/configs/config.js]: I'm not sure how to properly handle the config data considering there is no way to setup a local jsapi testing instance. I am going to add it to the gitignore for now.

Comment (jgpacheco) [inline spec/support/application/configs/config.js]: We can create a stubs for submodules. Check [proxyquire](https://www.npmjs.com/package/proxyquire) we are using this already in FC.

File: package.json (hunk 1 above)
Comment (rossnelson) [inline package.json]: Bash is universal no need for gulp

https://medium.freecodecamp.com/why-i-left-gulp-and-grunt-for-npm-scripts-3d6853dd22b8#.kllkwy88j

**Pull request merged**